wire-library/wire-runtime-swift/src/main/swift/Codable/Codable+Wire.swift

```diff
@@ -72,4 +49,114 @@ extension KeyedDecodingContainer {
         }
         return value
     }
+
+
+    public func decodeFirst<T : Decodable>(
+        _ type: T.Type,
+        forKeys firstKey: Key,
+        _ additionalKeys: Key...
+    ) throws -> T {
+        try decodeFirst(type, firstKey: firstKey, additionalKeys: additionalKeys)
+    }
+
+    public func decodeFirstIfPresent<T : Decodable>(
+        _ type: T.Type,
+        forKeys firstKey: Key,
+        _ additionalKeys: Key...
+    ) throws -> T? {
+        try decodeFirstIfPresent(type, firstKey: firstKey, additionalKeys: additionalKeys)
+    }
+}
+
+// MARK: - decodeStringEncoded()
+
+extension KeyedDecodingContainer {
+#warning("Revisit generics")
+
+    public func decodeStringEncoded<T: StringCodable & Codable>(
+        _ type: T.Type,
+        forKey key: Key
+    ) throws -> T {
+        try decode(StringEncoded<T>.self, forKey: key).wrappedValue
+    }
+
+    public func decodeStringEncodedIfPresent<T: StringCodable & Codable>(
+        _ type: T.Type,
+        forKey key: Key
+    ) throws -> T? {
+        try decodeIfPresent(StringEncoded<T>.self, forKey: key)?.wrappedValue
+    }
+
+    public func decodeFirstStringEncoded<T : StringCodable & Codable>(
+        _ type: T.Type,
+        forKeys firstKey: Key,
+        _ additionalKeys: Key...
+    ) throws -> T {
+        try decodeFirst(StringEncoded<T>.self, firstKey: firstKey, additionalKeys: additionalKeys).wrappedValue
+    }
+
+    public func decodeFirstStringEncodedIfPresent<T : StringCodable & Codable>(
+        _ type: T.Type,
+        forKeys firstKey: Key,
+        _ additionalKeys: Key...
+    ) throws -> T? {
+        try decodeFirstIfPresent(StringEncoded<T>.self, firstKey: firstKey, additionalKeys: additionalKeys)?.wrappedValue
+    }
+}
+
+// MARK: - decodeStringEncodedValues()
+
+extension KeyedDecodingContainer {
+    public func decodeStringEncodedValues<T : StringCodable & Codable>(
+        _ type: T.Type,
+        forKey key: Key
+    ) throws -> [T] {
+        try decode(StringEncodedValues<Array<T>>.self, forKey: key).wrappedValue
+    }
+
+    public func decodeStringEncodedValuesIfPresent<T : StringCodable & Codable>(
```

Comment:
_in theory_ we don't need both IfPresent and not IfPresent versions of this method.
That said, if we ever wanted to model a `required` repeated value, we'd need to be able to differentiate.

Reply:
Yeah, no such thing. `optional`, `required`, and `repeated` are all mutually exclusive values for the optionality modifier. There's no such concept as "a repeated field that contains at least one value"; put another way, an empty `repeated` field is the same as an unset `optional`.

In the encoded bytes, `repeated` fields aren't even represented as a collection. They're represented as the same single field encoded over and over, and if they never show up then the client is supposed to synthesize an empty array. (There's an optimized repeated packing format, but it's optional.)